### PR TITLE
Feature/nlp integration

### PR DESCRIPTION
**Pull request by bimal125**
Depends on https://github.com/the-deep/deeper/pull/974
Depends on https://github.com/the-deep/deepl-deep-integration/pull/1

Also, includes Unified connector

## Changes

* Add endpoints to send prediction request to nlp and handle callback 

Mention related users here if any.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/assisted_tagging/models.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+
+class AssistedTaggingModel(models.Model):
+    # This is for refering model id within deep. This can change. Source is the deepl.
+    class ModelID(models.TextChoices):
+        MAIN = 'all_tags_model', 'All tags model'
+        GEO = 'geolocation', 'Geo Location'
+        RELIABILITY = 'reliability', 'Reliability'
+
+    model_id = models.CharField(max_length=256)
```
Comment (k9845) [inline apps/assisted_tagging/models.py:21]: Won't this be choices?

File: apps/assisted_tagging/serializers.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+class WrongPredictionReviewGqlSerializer(UserResourceSerializer, serializers.ModelSerializer):
+    class Meta:
+        model = WrongPredictionReview
+        fields = (
+            'prediction',
+        )
+
+    def validate_prediction(self, prediction):
+        if prediction.draft_entry.project != self.context['request'].active_project:
+            raise serializers.ValidationError('Prediction not part of the active project.')
```
Comment (k9845) [inline apps/assisted_tagging/serializers.py:128]: May be add `Prediction Entry Project not part of the active project.`

File: apps/assisted_tagging/serializers.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        )
+
+    def validate_draft_entry(self, draft_entry):
+        if draft_entry.project != self.context['request'].active_project:
+            raise serializers.ValidationError('Draft Entry not part of the active project.')
+        return draft_entry
+
+    def validate(self, data):
+        if self.instance and self.instance.created_by != self.context['request'].user:
+            raise serializers.ValidationError('Only reviewer can edit this review')
```
Comment (k9845) [inline apps/assisted_tagging/serializers.py:153]: This is used in many classes lets make a common function.

File: apps/assisted_tagging/tasks.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+            # Refetch
+            current_tags_map = get_tags_map()
+            sync_tags_with_deepl.delay()
+        return current_tags_map
+
+    @classmethod
+    def _process_model_preds(cls, model_version, current_tags_map, draft_entry, model_prediction):
+        prediction_status = model_prediction['prediction_status']
+        draft_entry.prediction_status = DraftEntry.PredictionStatus.DONE
+        if prediction_status == 0:  # If 0 no tags are provided
```
Comment (k9845) [inline apps/assisted_tagging/tasks.py:182]: Why are we hard coding prediction_status, don't we have enum for this?

File: apps/assisted_tagging/tasks.py [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+            return
+
+        tags = model_prediction.get('tags', {})  # NLP TagId
+        values = model_prediction.get('values', [])  # Raw value
+        print(model_prediction.keys())
```
Comment (k9845) [inline apps/assisted_tagging/tasks.py]: Why!!!

File: apps/assisted_tagging/tasks.py [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+        ])
+        models_version_map = cls.get_or_create_models_version([
+            predition['model_info']
+            for predition in model_preds
+        ])
+
+        with transaction.atomic():
+            draft_entry.clear_data()  # Clear old data if exists
+            draft_entry.calculated_at = timezone.now()
+            for predition in model_preds:
```
Comment (k9845) [inline apps/assisted_tagging/tasks.py]: `prediction`

File: apps/assisted_tagging/models.py (hunk 1 above)
Comment (thenav56) [reply apps/assisted_tagging/models.py:21]: Nope. This is provided by deepl services. Enum used above is just for custom static behavior.

File: apps/assisted_tagging/serializers.py (hunk 2 above)
Comment (thenav56) [reply apps/assisted_tagging/serializers.py:128]: For users, there is no draft entry, only prediction.

File: apps/assisted_tagging/serializers.py (hunk 3 above)
Comment (thenav56) [reply apps/assisted_tagging/serializers.py:153]: Let's ignore this for now.

File: apps/assisted_tagging/tasks.py (hunk 4 above)
Comment (thenav56) [reply apps/assisted_tagging/tasks.py:182]: This is provided by the deepl.

File: apps/assisted_tagging/tasks.py (hunk 6 above)
Comment (thenav56) [reply apps/assisted_tagging/tasks.py]: Done

File: apps/assisted_tagging/tasks.py (hunk 5 above)
Comment (thenav56) [reply apps/assisted_tagging/tasks.py]: Done

**Review by k9845 — APPROVED** [review]

**Pull request merged**